app/views/layouts/application.html.erb

```diff
@@ -1,15 +1,43 @@
 <!DOCTYPE html>
 <html>
-<head>
-  <title>EbayClone</title>
-  <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
-  <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
-  <%= csrf_meta_tags %>
-  <%= stylesheet_link_tag "https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" %>
-</head>
-<body>
-
-<%= yield %>
-
-</body>
+  <head>
+    <title>EbayClone</title>
+    <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
+    <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
+    <%= csrf_meta_tags %>
+    <%= stylesheet_link_tag "https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" %>
+  </head>
+  <body>
+
+    <nav class="navbar navbar-default">
+      <div class="container-fluid">
+        <div class="navbar-header">
+          <a class="navbar-brand" href="#">Ebay Clone</a>
+        </div>
+
+        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
+          <% if session[:user_id] %>
+            <ul class="nav navbar-nav">
+              <li class="active"> <%= link_to("Log out", logout_path) %></li>
+            </ul>
+          <% else %>
+            <ul class="nav navbar-nav">
+              <li class="active"> <%= link_to("Log in", login_path) %></li>
+            </ul>
+          <% end %>
+          <ul class="nav navbar-nav">
+            <li class="active"> <%= link_to("Item Marketplace", items_path) %></li>
+          </ul>
+          <% if current_user %>
+            <ul class="nav navbar-nav navbar-right">
+              <li class="navbar-text">Hello <%= current_user.username.capitalize %></li>
+            </ul>
+          <% end %>
```

Comment:
At the risk of having a few similarities I would prefer to have 2 partials, one for the logged in user and one for anonymous user.